### PR TITLE
Migrate to a secret store by default

### PR DESCRIPTION
**Pull request by benmoriceau**
## What
This is adding a metadata entry in order to make the use of the DB secret store to be the default one.
It will avoid having secret outside of the secret table.

## File to look

- `EnvConfig` => make the use of the DB secret store to be the default one
- `JobPersistence` => Access and update of the metadata access
- `BootloaderApp` => Change the logic to run or not the migration

### CODE REVIEW

File: airbyte-bootloader/src/test/java/io/airbyte/bootloader/SecretMigratorTest.java [hunk 1]
```diff
@@ -43,11 +44,14 @@ public class SecretMigratorTest {
   @Mock
   private SecretPersistence secretPersistence;
 
+  @Mock
+  private JobPersistence jobPersistence;
```
Comment (lmossman) [inline airbyte-bootloader/src/test/java/io/airbyte/bootloader/SecretMigratorTest.java:48]: Should probably add unit tests here for the different cases of `jobPersistence.isSecretMigrated()`

File: docker-compose.yaml [hunk 2]
```diff
@@ -48,6 +48,8 @@ services:
       - POSTGRES_USER=${DATABASE_USER}
     volumes:
       - db:/var/lib/postgresql/data
+    ports:
+      - 5432:5432
```
Comment (lmossman) [inline docker-compose.yaml:52]: Just double checking - this is something you want to commit to master, right?

**Review by lmossman — APPROVED** [review]
Couple of small comments, otherwise LGTM

File: docker-compose.yaml (hunk 2 above)
Comment (benmoriceau) [reply docker-compose.yaml:52]: No it is not, I am using that to be able to query my local DB when running docker-compose but I forget to remove it.

File: airbyte-bootloader/src/test/java/io/airbyte/bootloader/SecretMigratorTest.java (hunk 1 above)
Comment (benmoriceau) [reply airbyte-bootloader/src/test/java/io/airbyte/bootloader/SecretMigratorTest.java:48]: Will do.

Comment (benmoriceau) [reply airbyte-bootloader/src/test/java/io/airbyte/bootloader/SecretMigratorTest.java:48]: @lmossman I have added more tests in the `BootloaderAppTest` for that.

Comment (lmossman) [reply airbyte-bootloader/src/test/java/io/airbyte/bootloader/SecretMigratorTest.java:48]: 👍 Looks good

**Pull request merged**